src/main/java/com/yelp/nrtsearch/server/config/IndexStartConfig.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.config;
+
+import com.yelp.nrtsearch.server.grpc.Mode;
+import java.util.Objects;
+
+/** Configuration class for starting indices from global state. */
+public class IndexStartConfig {
+  public static final String CONFIG_PREFIX = "indexStartConfig.";
+  public static final String PRIMARY_DISCOVERY_PREFIX = "primaryDiscovery.";
+
+  public enum IndexDataLocationType {
+    LOCAL,
+    REMOTE
+  }
+
+  private final boolean autoStart;
+  private final Mode mode;
+  private final String discoveryHost;
+  private final Integer discoveryPort;
+  private final String discoveryFile;
+  private final IndexDataLocationType dataLocationType;
+
+  /**
+   * Create instance from provided configuration reader.
+   *
+   * @param configReader config reader
+   * @return class instance
+   */
+  public static IndexStartConfig fromConfig(YamlConfigReader configReader) {
+    Objects.requireNonNull(configReader);
+    boolean autoStart = configReader.getBoolean(CONFIG_PREFIX + "autoStart", false);
+    Mode mode = Mode.valueOf(configReader.getString(CONFIG_PREFIX + "mode", "STANDALONE"));
+    String discoveryHost =
+        configReader.getString(CONFIG_PREFIX + PRIMARY_DISCOVERY_PREFIX + "host", "");
+    Integer discoveryPort =
+        configReader.getInteger(CONFIG_PREFIX + PRIMARY_DISCOVERY_PREFIX + "port", 0);
+    String discoveryFile =
+        configReader.getString(CONFIG_PREFIX + PRIMARY_DISCOVERY_PREFIX + "file", "");
```

Comment:
what if all of host, port and file are missing? Do we have any check about this?

Reply:
This condition is caught during the index start process https://github.com/aprudhomme/nrtsearch/blob/auto_start_indices/src/main/java/com/yelp/nrtsearch/server/luceneserver/StartIndexHandler.java#L175 I wanted to avoid strict validation here since it is still an optional feature.